File: teams/script-money/week4_work.py
from flow_py_sdk import Script, flow_client
import json
import asyncio


async def get_nba_transaction_events():
    async with flow_client(
        host="access.mainnet.nodes.onflow.org", port=9000
    ) as client:
        block = await client.get_latest_block(is_sealed=True)
        print(f"block.height: {block.height}\n")
        startBlock, endBlock = block.height-10, block.height
        event_type = "A.c1e4f4f4c4257510.Market.MomentPurchased"
        block_events = await client.get_events_for_height_range(
            type=event_type,
            start_height=startBlock,
            end_height=endBlock
        )
        for block_event in block_events:
            for event in block_event.events:
                print(f"type: {event.type}")
                print(f"transaction_id: {event.transaction_id.hex()}")
                fields = json.loads(event.payload)['value']['fields']
                id = fields[0]['value']['value']
                print(f"id: {int(id)}")
                price = fields[1]['value']['value']
                print(f"price: {float(price)}")
                seller = fields[2]['value']['value']['value']
                print(f"seller: {seller}")
                print('\n')

if __name__ == "__main__":
    asyncio.run(get_nba_transaction_events())

File: teams/Skyone/week4_work.go
package main

import (
 "context"
 "fmt"
 "github.com/onflow/flow-go-sdk/client"
 "google.golang.org/grpc"
)

func main() {
 ctx := context.Background()

 flowClient, err := client.New("access.mainnet.nodes.onflow.org:9000", grpc.WithInsecure())
 if err != nil{
  panic(err)
 }

 blockHeader, err := flowClient.GetLatestBlockHeader(ctx, true)
 if err != nil{
  panic(err)
 }

 blockTip := blockHeader.Height
 fmt.Printf("Current block tip: ", blockHeader.Height)

 startBlock, endBlock := blockTip-20, blockTip
 const eventType = "A.c1e4f4f4c4257510.Market.MomentPurchased"
 results, err := flowClient.GetEventsForHeightRange(ctx, client.EventRangeQuery{
  Type:        eventType,
  StartHeight: startBlock,
  EndHeight:   endBlock,
 })
 if err != nil{
  panic(err)
 }

 fmt.Println("Query result of NBA Topshot event: ",eventType" from block ",startBlock, " to ",endBlock)
 for _, block := range results {
  for i, event := range block.Events {
   fmt.Println("Found event ",i+1," in block ", block.Height)
   fmt.Println("Transaction ID: ", event.TransactionID)
   fmt.Println("Event ID: ", event.ID())
   fmt.Println("Moment ID: ", event.Value.Fields[0])
   fmt.Println("Price(int): ", event.Value.Fields[1].ToGoValue())
   fmt.Println("Price(str): ", event.Value.Fields[1].String())
   fmt.Println("Seller: ", event.Value.Fields[2])
   fmt.Println(event.Value)
  }
 }
}
